lib/msal-common/docs/claims.md

```diff
@@ -0,0 +1,21 @@
+# Claims and Claims Challenges in MSAL
+
+A resource server may deem an access token it receives as having insufficient claims. If the client application that sends the access token has declared itself to be capable, the resource server may respond with a claims challenge, upon which the client can acquire a new token satisfying the claims challenge and try again.
```

Comment:
I'm under the impression that an RP can respond to a claims challenge even if the client irrespective of the capabilies of the client. CC @jmprieur

Reply:
MSAL py docs seem more accurate to me: 

`Client capability is meant to inform the Microsoft identity platform (STS) what this client is capable for, so STS can decide to turn on certain features. For example, if client is capable to handle claims challenge, STS can then issue CAE access tokens to resources knowing when the resource emits claims challenge the client will be capable to handle.
`

I think AAD only supports "CP1" right now, which is what you described, but the mechanism is more general purpose.

Reply:
Makes sense. I've break down the content into separate pages, with cross links to each other.